ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ActiveContractsService.scala

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.tests
+
+import com.daml.ledger.api.testtool.infrastructure.{
+  LedgerSession,
+  LedgerTest,
+  LedgerTestContext,
+  LedgerTestSuite
+}
+import com.digitalasset.ledger.api.v1.event.Event.Event.Created
+import com.digitalasset.ledger.api.v1.event.{CreatedEvent, Event}
+import com.digitalasset.ledger.client.binding.Primitive.{Party, TemplateId}
+import com.digitalasset.ledger.test.Test.Dummy._
+import com.digitalasset.ledger.test.Test.{Dummy, DummyFactory, DummyWithParam}
+import io.grpc.Status
+import scalaz.syntax.tag._
+
+class ActiveContractsService(session: LedgerSession) extends LedgerTestSuite(session) {
+  private val invalidLedgerId =
+    LedgerTest(
+      "ACSinvalidLedgerId",
+      "The ActiveContractService should fail for requests with an invalid ledger identifier") {
+      ledger =>
+        val invalidLedgerId = "ACSinvalidLedgerId"
+        for {
+          party <- ledger.allocateParty()
+          invalidRequest = ledger
+            .activeContractsRequest(Seq(party))
+            .update(_.ledgerId := invalidLedgerId)
+          failure <- ledger.activeContracts(invalidRequest).failed
+        } yield {
+          assertGrpcError(failure, Status.Code.NOT_FOUND, "not found. Actual Ledger ID")
```

Comment:
Maybe not worth doing, but you can add some extra strictness to this test checking that the reported invalid ledger identifier matches the one provided.